estate/models/estate_property.py

```diff
@@ -0,0 +1,138 @@
+        string="State",
+        selection=[
+            ('draft', "Draft"),
+            ('new', "New"),
+            ('offer_received', "Offer Received"),
+            ('offer_accepted', "Offer Accepted"),
+            ('sold', "Sold"),
+            ('cancelled', "Cancelled")
+        ],
+        copy=False,
+        default='new',
+        tracking=True,
+    )
+    property_type_id = fields.Many2one('estate.property.type', string="Property Type")
+    salesperson_id = fields.Many2one('res.partner', string="Salesperson", tracking=True)
+    buyer_id = fields.Many2one('res.partner', string="Buyer", copy=False, tracking=True)
+    tag_ids = fields.Many2many('estate.property.tag', string="Property Tags")
+    offer_ids = fields.One2many(
+        'estate.property.offer',
+        'property_id',
+        string="Property Offers",
+    )
+    company_id = fields.Many2one('res.company', string="Company", default=lambda self: self.env.company, required=True)
+    total_area = fields.Float(compute='_compute_total_area', store=True)
+    best_price = fields.Float(compute='_compute_best_price', store=True)
+    image_1920 = fields.Image(string="Property Image", max_width=1920, max_height=1920)
+
+    _sql_constraints = [
+        ('positive_expected_price', 'CHECK(expected_price > 0)', 'A property expected price must be strictly positive.'),
+        ('positive_selling_price', 'CHECK(selling_price > 0)', 'A property selling price must be positive.'),
+    ]
+
+    @api.model
+    def ondelete(self):
+        for property in self:
+            if property.state not in ['new', 'cancelled']:
+                raise UserError(f"Cannot delete property in state '{property.state}'. It can only be deleted if it is 'New' or 'Cancelled'.")
+
+    @api.constrains('expected_price', 'selling_price')
+    def _check_selling_price(self):
+       for property in self:
+          if float_compare(property.selling_price, 0.0, precision_digits=2) == 0 and not property.offer_ids.filtered(lambda offer: offer.state == 'offer accepted'):
+              continue
+          if float_compare(property.selling_price, property.expected_price * 0.9, precision_digits=2) < 0:
+               raise ValidationError(
+                 ("The selling price cannot be lower than 90 percentage of the expected price. Expected Price: %.2f, Selling Price: %.2f") %
+                 (property.expected_price, property.selling_price)
+                )
+
+    @api.depends('living_area', 'garden_area')
+    def _compute_total_area(self):
+       for property in self:
+          property.total_area = property.living_area + property.garden_area
+
+    @api.depends('offer_ids')
+    def _compute_best_price(self):
+       for property in self:
+             property.best_price = max(property.offer_ids.mapped('price'), default=0.0)
+
+    @api.onchange('garden')
+    def _onchange_garden(self):
+       if self.garden:
+        self.garden_area, self.garden_orientation = 10, 'north'
```

Comment:
```suggestion
        self.garden_area = 10
        self.garden_orientation = 'north'
```
do this separately or with a single CRUD operation